packages/fxa-settings/src/components/FormPasswordWithInlineCriteria/index.tsx

```diff
@@ -0,0 +1,382 @@
+        'Entered passwords match.'
+      );
+      setSROnlyConfirmPwdFeedbackMessage(srOnlyPasswordsMatch);
+    }
+
+    if (!formState.isValid) {
+      trigger('newPassword');
+    }
+  }, [
+    formState,
+    ftlMsgResolver,
+    getValues,
+    hideConfirmPwdInfo,
+    isConfirmPwdRequirementsVisible,
+    localizedPasswordMatchError,
+    passwordFormType,
+    setPasswordMatchErrorText,
+    trigger,
+  ]);
+
+  const onChangePassword = (inputName: string) => {
+    const newPassword = getValues('newPassword');
+    const confirmPassword = getValues('confirmPassword');
+    if (inputName === 'newPassword') {
+      !hasUserTakenAction && setHasUserTakenAction(true);
+      trigger('newPassword');
+    }
+
+    if (!errors.newPassword) {
+      setSROnlyPwdFeedbackMessage('');
+    }
+
+    if (confirmPassword !== newPassword && confirmPassword !== '') {
+      setPasswordMatchErrorText(localizedPasswordMatchError);
+    } else {
+      setPasswordMatchErrorText('');
+    }
+
+    trigger('confirmPassword');
+  };
+
+  return (
+    <>
+      <form {...{ onSubmit }} className="flex flex-col">
+        {/* Hidden email field is to help password managers
+           correctly associate the email and password. Without this,
+           password managers may try to use another field as username */}
+        <input
+          type="email"
+          value={email}
+          className="hidden"
+          autoComplete="username"
+          readOnly
+        />
+        <div>
+          <span
+            id="password-requirements"
+            aria-live="polite"
+            className="text-xs"
+          >
+            <PasswordStrengthInline
+              {...{
+                hasUserTakenAction,
+                isTooShort: errors.newPassword?.types?.length,
+                isSameAsEmail: errors.newPassword?.types?.notEmail,
+                isCommon: errors.newPassword?.types?.uncommon,
+                isUnconfirmed:
+                  getValues('confirmPassword') !== getValues('newPassword'),
+              }}
+            />
+            {srOnlyPwdFeedbackMessage && (
+              <span className="sr-only">{srOnlyPwdFeedbackMessage}</span>
+            )}
+          </span>
+        </div>
+
+        <div className="relative mb-4" aria-atomic="true">
+          <FtlMsg id={templateValues.passwordFtlId} attrs={{ label: true }}>
+            <InputPassword
+              name="newPassword"
+              label={templateValues.passwordLabel}
+              onFocusCb={onNewPwdFocus}
+              onBlurCb={onNewPwdBlur}
+              onChange={() => onChangePassword('newPassword')}
+              hasErrors={
+                formState.dirtyFields.newPassword ? errors.newPassword : false
+              }
+              inputRef={register({
+                required: true,
+                validate: {
+                  // TODO in FXA-7482, review our password requirements and best way to display them
```

Comment:
This ticket was closed with no change recommended. If we'd like to recommend another review, we should open a new ticket.